core/common/src/main/java/alluxio/metrics/MetricKey.java

```diff
@@ -977,6 +977,294 @@ public MetricKey build() {
           .setMetricType(MetricType.COUNTER)
           .build();
 
+  // Rocks Block metrics
+  public static final MetricKey MASTER_ROCKS_BLOCK_BACKGROUND_ERRORS =
+      new Builder("Master.RocksBlockBackgroundErrors")
+          .setDescription("RocksDB block table. Accumulated number of background errors.")
+          .setMetricType(MetricType.GAUGE)
+          .build();
+  public static final MetricKey MASTER_ROCKS_BLOCK_BLOCK_CACHE_CAPACITY =
+      new Builder("Master.RocksBlockBlockCacheCapacity")
+          .setDescription("RocksDB block table. Block cache capacity.")
+          .setMetricType(MetricType.GAUGE)
+          .build();
+  public static final MetricKey MASTER_ROCKS_BLOCK_BLOCK_CACHE_PINNED_USAGE =
+      new Builder("Master.RocksBlockBlockCachePinnedUsage")
+          .setDescription("RocksDB block table. Memory size for the entries being pinned.")
+          .setMetricType(MetricType.GAUGE)
+          .build();
+  public static final MetricKey MASTER_ROCKS_BLOCK_BLOCK_CACHE_USAGE =
+      new Builder("Master.RocksBlockBlockCacheUsage")
+          .setDescription("RocksDB block table. Memory size for the entries residing in block "
+              + "cache.")
+          .setMetricType(MetricType.GAUGE)
+          .build();
+  public static final MetricKey MASTER_ROCKS_BLOCK_COMPACTION_PENDING =
```

Comment:
What does the user do if `1` is seen? What does the user do if `0` is seen?

Reply:
https://github.com/facebook/rocksdb/wiki/RocksDB-Overview#multi-threaded-compactions

I'm not sure what they are supposed to do, or if there is anything to do. It's more informational than anything.

Reply:
Personally I am OK with exposing any or all of RocksDB metrics, as long as the user has a link back to what they are in the RocksDB GitHub. I suppose there could be some way to tag certain metrics as "for advanced users", but maybe that should be considered as a separate task. There is also the question of performance overhead of having lots of metrics, but that should also be considered separately as something that should be done.

Reply:
Maybe in the MetricKey ``Builder.setDecription`` you can include the name of the RocksDB property that the metric is associated with, ie. ``rocksdb.compaction-pending`` for this one

Reply:
I believe if we don't understand a certain metric ourselves, exposing that is just asking for questions we cannot answer. Especially if a customer is asking about a metric, how do we just throw them to the RocksDB doc page?

@yuzhu My baseline is, we can expose a metric only if we know how to answer a "What is this" question. WDYT?

Reply:
I think it is ok to expose it as long as we make it clear it corresponds to a rocksdb metric. In our doc, we should just clearly state rocksdb.* are rocks internal metrics and only explain a few that we care about.

Reply:
@yuzhu Ok, but by seeing these metrics, users will immediately assume we are supporting those and will have expectations. As long as we are ready to handle that in the near future(before 2.9), I've no problem merging the code.